### ISSUE TITLE
Inconsistency logic behavior between mergeWith & mergeDeepWith

### ISSUE THREAD

**Issue by vpcvdc** [immutable-js/immutable-js#1774]
### What happened

mergeWith working fine as per https://immutable-js.github.io/immutable-js/docs/#/mergeWith docs.
mergeDeepWith NOT working fine as per https://immutable-js.github.io/immutable-js/docs/#/mergeDeepWith.
It says like mergeDeepWith is recursive implementation of mergeWith. @leebyron Isn't it?

### How to reproduce

<img width="1680" alt="Screen Shot 2020-07-07 at 12 05 50 PM" src="https://user-images.githubusercontent.com/17532117/86729220-dc0a5900-c04a-11ea-9e4f-3fedf3985975.png">

**Comment by Methuselah96** [comment]
@vpcvdc Can you expound on what you think the inconsistent behavior is? It looks like it is working correctly to me.

The first example only goes one layer deep, so it ends up calling `console.log([1], [1, 2, 3]);`.

The second and third examples go all the way to the deepest layer and since arrays get merged together, they end up calling `console.log([1, 1, 2, 3])`.

**Comment by vpcvdc** [comment]
Hi @Methuselah96

As per document says 

> calling the merger function whenever an existing value is encountered.

In the above screenshot, `merger => console.log`
mergeWith function calling merger function when an exist value encountered on particular property key
But the same thing **NOT** happening on mergeDeepWith 🤯

In the above screenshot, mergeDeepWith behaving like mergeDeep 😳

I hope you get the clarity, what I really mean 🙂

**Comment by Methuselah96** [comment]
@vpcvdc I don't think that arrays are considered values in this case. The merger function isn't called on objects or arrays **EXCEPT** in the case where we're more than one layer deep in `mergeWith`.

For example the merger function is called on objects here (similar to how they were called on arrays in your first example):
![image](https://user-images.githubusercontent.com/693755/90401845-8aeb7d00-e06c-11ea-81aa-ba7ad01d52f1.png)

**Comment by conartist6** [comment]
I'm trying to pick apart what's going on here.

To start with the docs are wrong. There is no such method as `Collection.prototype.mergeWith`. There is only `Map.prototype.mergeWith`. This explains why the merger is never called on arrays or Lists.

To fix this we need to either remove `Immutable.mergeWith` or implement `Collection.prototype.mergeWith`.

And of course the situation is the same for `mergeDeepWith`.
